drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,723 @@
+        }
+        break;
+    }
+
+    sx127x_set_state(dev, SX127X_RF_RX_RUNNING);
+
+    if (dev->settings.lora.rx_continuous) {
+        sx127x_set_op_mode(dev, SX127X_RF_LORA_OPMODE_RECEIVER);
+    }
+    else {
+        if (dev->settings.window_timeout != 0) {
+            xtimer_set(&(dev->_internal.rx_timeout_timer),
+                       dev->settings.window_timeout);
+        }
+        sx127x_set_op_mode(dev, SX127X_RF_LORA_OPMODE_RECEIVER_SINGLE);
+    }
+}
+
+void sx127x_set_max_payload_len(sx127x_t *dev, uint8_t maxlen)
+{
+    sx127x_radio_modems_t modem = dev->settings.modem;
+    switch (modem) {
+        case SX127X_MODEM_FSK:
+            break;
+
+        case SX127X_MODEM_LORA:
+            sx127x_reg_write(dev, SX127X_REG_LR_PAYLOADMAXLENGTH, maxlen);
+            break;
+    }
+}
+
+uint8_t sx127x_get_op_mode(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_OPMODE) & ~SX127X_RF_OPMODE_MASK;
+}
+
+void sx127x_set_op_mode(sx127x_t *dev, uint8_t op_mode)
+{
+    static uint8_t op_mode_prev = 0;
+
+    op_mode_prev = sx127x_get_op_mode(dev);
+
+    if (op_mode != op_mode_prev) {
+        /* Replace previous mode value and setup new mode value */
+        sx127x_reg_write(dev, SX127X_REG_OPMODE,
+                         (op_mode_prev & SX127X_RF_OPMODE_MASK) | op_mode);
+    }
+}
+
+sx127x_lora_bandwidth_t sx127x_get_bandwidth(sx127x_t *dev)
+{
+    return dev->settings.lora.bandwidth;
+}
+
+inline void _low_datarate_optimize(sx127x_t *dev)
+{
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+        if ( ((dev->settings.lora.bandwidth == SX127X_BW_125_KHZ) &&
+              ((dev->settings.lora.datarate == SX127X_SF11) ||
+               (dev->settings.lora.datarate == SX127X_SF12))) ||
+             ((dev->settings.lora.bandwidth == SX127X_BW_250_KHZ) &&
+              (dev->settings.lora.datarate == SX127X_SF12))) {
+            dev->settings.lora.low_datarate_optimize = 0x01;
+        } else {
+            dev->settings.lora.low_datarate_optimize = 0x00;
+        }
+
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG3,
+                         (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG3) &
+                          SX127X_RF_LORA_MODEMCONFIG3_LOWDATARATEOPTIMIZE_MASK) |
+                         (dev->settings.lora.low_datarate_optimize << 3));
+        break;
+    case SX1272_TYPE:
+        if ( (dev->settings.lora.bandwidth == SX127X_BW_125_KHZ) &&
+             ((dev->settings.lora.datarate == SX127X_SF11) ||
+              (dev->settings.lora.datarate == SX127X_SF12))) {
+            dev->settings.lora.low_datarate_optimize = 0x01;
+        } else {
+            dev->settings.lora.low_datarate_optimize = 0x00;
+        }
+
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1,
+                         (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1) &
+                          SX127X_RF_LORA_MODEMCONFIG1_LOWDATARATEOPTIMIZE_MASK) |
```

Comment:
Aaaaah, this one gives me a huge headache....
Currently `SX127X_RF_LORA_MODEMCONFIG1_LOWDATARATEOPTIMIZE_MASK` is 0x01 so this call clear all the bits instead of protect them. This mask must be 0xFE to do the correct job.
This bug prevents SX1272 to set a bandwidth superior to 125 kHz...
You can check it using register get 0x1D shell command before and after setup a new bandwidth.